resteasy-core/src/main/java/org/jboss/resteasy/core/ContextParameterInjector.java

```diff
@@ -39,6 +43,7 @@
    private ResteasyProviderFactory factory;
    private Type genericType;
    private Annotation[] annotations;
+   private boolean outputStreamWasWritten = false;
```

Comment:
This should at least be `volatile` if not even an `AtomicBoolean` or protected by something. Unless of course we know for sure the `ContextParameterInjector` instance is used by a single thread at a time. I do not know that :)

Reply:
Yep, there's no reason that ContextParameterInjector couldn't be shared among threads.